### PR TITLE
feat(cl_scrape_opinions): ingest more Juriscraper fields

### PR DESCRIPTION
**Pull request by grossir**
Partially solves #4042

- Ingest "lower_courts" into `Docket.appeal_from_str`
- Ingest "dispositions" into `OpinionCluster.disposition`
- Ingest "other_dates" into `OpinionCluster.other_dates`
- Ingest "authors" into `Opinion.author_str`
- Ingest "joined_by" into `Opinion.joined_by`
- Ingest "per_curiam" into `Opinion.per_curiam`
- Ingest "types" into `Opinion.type`

Last 4 fields are not supported in Juriscraper as of yet, but the changes proposed keep the default behavior

### CODE REVIEW

**Comment by grossir** [comment]
I tested the ingestion of the new fields by running the `md` scraper on the main branch, which returns "per_curiam" and "other_dates"

```
select * from search_opinioncluster where docket_id in (select id from search_docket where court_id='md') and other_dates <> '';


-[ RECORD 3 ]-------------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
--------------------
id                        | 86
date_created              | 2024-07-18 23:55:59.377911+00
date_modified             | 2024-07-18 23:56:01.080969+00
judges                    | Booth
date_filed                | 2024-07-09
date_filed_is_approximate | f
slug                      | town-of-bel-air-v-bodt
case_name_short           | Bodt
case_name                 | Town of Bel Air v. Bodt
case_name_full            | 
scdb_id                   | 
scdb_decision_direction   | 
scdb_votes_majority       | 
scdb_votes_minority       | 
source                    | C
procedural_history        | 
attorneys                 | 
nature_of_suit            | 
posture                   | 
syllabus                  | 
headnotes                 | 
summary                   | 
disposition               | 
history                   | 
other_dates               | corrected 2027-07-15
cross_reference           | 
correction                | 
citation_count            | 0
precedential_status       | Published
date_blocked              | 
blocked                   | f
filepath_json_harvard     | 
docket_id                 | 86
arguments                 | 
headmatter                | 
```

```
select * from search_opinion where cluster_id in (select id from search_opinioncluster where docket_id in (select id from search_docket where court_id='md'));

id                  | 90
date_created        | 2024-07-18 23:56:53.863707+00
date_modified       | 2024-07-18 23:56:56.101204+00
author_str          | 
per_curiam          | t
joined_by_str       | 
type                | 010combined
sha1                | afed8ab65c5e44486800a2b0b0659b466962d7e0
page_count          | 15
download_url        | https://www.mdcourts.gov/data/opinions/coa/2024/25a23agpc.pdf
local_path          | pdf/2024/07/05/attorney_grievance_commn_v._mosby.pdf
plain_text          |                                                           IN THE SUPREME COURT                                                                                                               

```

And tested the `Docket.lower_court_str` ingestion by running bap1
```
courtlistener=# select date_created,court_id, appeal_from_str, case_name from search_docket where court_id='bap1';
         date_created          | court_id |              appeal_from_str              |        case_name        
-------------------------------+----------+-------------------------------------------+-------------------------
 2024-07-19 00:04:07.203778+00 | bap1     | USBC - District of Massachusetts (Boston) | Vincent Zarrilli v. IRS
 2024-07-19 00:04:21.010286+00 | bap1     | USBC - District of Massachusetts (Boston) | V&M Management v.

```

**Comment by grossir** [comment]
@flooie can you please check this one, it is finally passing tests

File: cl/scrapers/management/commands/cl_scrape_opinions.py [hunk 1]
```diff
@@ -165,14 +185,21 @@ def save_everything(
         citation.cluster_id = cluster.pk
         citation.save()
 
+    if opinion.author_str:
+        candidate = async_to_sync(lookup_judges_by_messy_str)(
+            opinion.author_str, docket.court.pk, cluster.date_filed
+        )
+        if candidate:
+            opinion.author = candidate[0]
```
Comment (flooie) [inline cl/scrapers/management/commands/cl_scrape_opinions.py:193]: Dont save if we find more than one judge please

Comment (grossir) [reply cl/scrapers/management/commands/cl_scrape_opinions.py:193]: @flooie I updated the PR and it is passing tests

**Comment by flooie** [comment]
Thanks @grossir this looks great.

**Comment by mlissner** [comment]
Nice on, @grossir. Glad to see our framework getting better!

**Pull request merged**